BaseClasses.py

```diff
@@ -542,7 +542,7 @@ def has_beaten_game(self, state: CollectionState, player: Optional[int] = None)
         else:
             return all((self.has_beaten_game(state, p) for p in range(1, self.players + 1)))
 
-    def can_beat_game(self, starting_state: Optional[CollectionState] = None) -> bool:
+    def can_beat_game(self, starting_state: Optional[CollectionState] = None, locations: Iterable[Location] = None) -> bool:
```

Comment:
```suggestion
    def can_beat_game(self,
                      starting_state: Optional[CollectionState] = None,
                      locations: Optional[Iterable[Location]] = None) -> bool:
```
This addresses 2 issues:
- The line length was over 120.
- pep 484 says this:
  > A past version of this PEP allowed type checkers to assume an optional type when the default value is None, as in this code:
  > 
  > `def handle_employee(e: Employee = None): ...`
  > 
  > This would have been treated as equivalent to:
  > 
  > `def handle_employee(e: Optional[Employee] = None) -> None: ...`
  > 
  > This is no longer the recommended behavior. Type checkers should move towards requiring the optional type to be made explicit.

Reply:
Oh yeah, true. Thank you! I threw this together in like 10 minutes so I'm surprised there aren't more problems like this lol